docs/references/global-shell.md

```diff
@@ -0,0 +1,153 @@
+So, if you have links saved that point to one of the older formats, those links will still work: the server will redirect those requests to the new URLs.
+
+#### Deep linking
+
+“Deep linking” still works, both with URLs using the new and old format. That is, you can save the URL to a particular dashboard, visualization, or data entry context selection, then use that link later to open that same view.
+
+### Example
+
+Take for example a request to `/apps/dashboard#/nghVC4wtyzi`, where `nghVC4wtyzi` is the ID of a dashboard. The server sees this request and routes it to the Global Shell, which is an app hosted at `/apps/`. The Global Shell loads and starts its routing process: it will construct a URL to load the Dashboard app in an `iframe` with any other URL information it received itself.
+
+It starts with determining the requested app: It sees `dashboard` as the path segment following `/apps/`, which it uses as the name of the requested app. It finds the Dashboard app in the server’s app list, and gets the Dashboard app’s launch path, `/dhis-web-dashboard/index.html`.
+
+It also forwards any hash and search values in the URL to the app it will load too, so the URL so far is `/dhis-web-dashboard/index.html#/nghVC4wtyzi`.
+
+Then the Global Shell will load the app with its search and hash in an `iframe`. To load the actual Dashboard app assets though, we also need to add `?redirect=false` to the URL’s search — by default, requests to the legacy app URL will get redirected to the Global Shell to load the app there. Since we’re already in the Global Shell, we don’t want to get redirected again.
+
+Therefore, the resulting URL is `/dhis-web-dashboard/index.html?redirect=false#/nghVC4wtyzi`, which is used as the `src` for an `iframe`, resulting in loading the Dashboard app with the correct dashboard.
+
+### Client-side routing and app routing
+
+The routing has somewhat of a two-way function: the top-level URL will determine what’s loaded, and the behavior of the app itself can affect the top-level URL.
+
+For example, loading `/apps/dashboard#/<dashboardId>` will load the dashboard app, and also pass the hash value to the Dashboard app, loading that particular dashboard in the app.
+
+In the reverse direction, if the user navigates to another Dashboard, the Global Shell can watch for that change inside the `iframe` and then update the hash value of the top level URL.
+
+### Bypasses
+
+If you want to view an app in “standalone” mode outside of the Global Shell, there are a few ways to do so.
+
+If you’re viewing an app in the Global Shell, i.e. at a URL like `/apps/dashboard`, you can append `?shell=false` to the search in the URL. The server will redirect that request to the legacy app URL, `/dhis-web-dashboard/?shell=false`, and you’ll view that app outside the Global Shell.
+
+If you have a legacy app URL, like `/dhis-web-dashboard/`, you can add either `?redirect=false` or `?shell=false` to the URL search to view it in standalone mode.
```

Comment:
are `shell=false` and `redirect=false` the same?

Reply:
Not exactly, which is why I kinda split up these paragraphs like this:
* `shell=false` and `redirect=false` _are_ the same if you're using a legacy app URL
* only `shell=false` works when added to a Global Shell URL, like `/apps/dashboard?shell=false`